File: Back to Basics Approach 3rd Edition/CodeStepByStep_java/if_else/printRange.java
package CodeStepByStep_java.if_else;

/**
 * https://www.codestepbystep.com/problem/view/java/ifelse/printRange
 *
 * @author Mr.Dr.Professor
 * @since 01-Mar-21 1:52 AM
 */
/*public*/ class printRange {
  public static void main(String[] args) {
  printRange(2, 7);
  printRange(19, 11);
  }

  public static void printRange (int a, int b) {
    if (a <= b) {
      System.out.print(a++);
      while (a <= b) {
        System.out.print(" " + a++);
      }
    } else {
      System.out.print(a--);
      while (a >= b) {
        System.out.print(" " + a--);
      }
    }
    System.out.println();
  }
}

File: Back to Basics Approach 3rd Edition/CodeStepByStep_java/javascript/recursion/printRange.js
/**
 * https://www.codestepbystep.com/problem/view/javascript/recursion/printRange
 *
 * @author Alex Golub
 * @since 09-May-21, 5:16 PM
 */
function printRange(x, y) {
  if (x > y) {
    throw 'x must be less than or equal to y';
  }
  console.log(helper(x, y));
}

function helper(x, y) {
  if (x === y) {
    return x;
  } else if (x + 1 === y) { // y is even
    return x + ' -- ' + y;
  } else {
    return `${x} > ${helper(x + 1, y - 1)} < ${y}`;
  }
}
